api/api/events/test_list_create_events.py

```diff
@@ -0,0 +1,173 @@
+    return APIClient()
+
+
+@pytest.mark.django_db
+def testListEvents_noEventsInDb_returnsEmptyList(test_user, client):
+    # given
+    client.force_authenticate(test_user)
+
+    # when
+    response = client.get('/events/', format='json')
+
+    # then
+    assert response.data['count'] == 0
+
+
+@pytest.mark.django_db
+def testListEvents_oneEventInDb_returnsOneEventInResponse(client, test_user, test_club):
+    # given
+    client.force_authenticate(test_user)
+
+    Events.objects.create(name='Introduction to Git and Github',
+                          description='some desc',
+                          starts_at='2022-08-07',
+                          ends_at='2022-08-07',
+                          image_url='https://asite.com',
+                          address='c',
+                          club=test_club)
+
+    # when
+    response = client.get('/events/', format='json')
+
+    # then
+    assert response.status_code == status.HTTP_200_OK
+    assert response.data['count'] == 1
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('position', ['Core member', 'Coordinator', 'Co-Coordinator'])
+def testCreateEvents_coreMemberRequest_creationSuccessful(client, test_user, test_club, position):
+    # given
+    client.force_authenticate(test_user)
+    Roles.objects.create(name=position,
+                         club=test_club,
+                         user=test_user,
+                         assigned_at='2022-08-01')
+
+    # when
+    response = client.post('/events/', {'name': 'Introduction to Git and Github',
+                                        'description': 'some desc',
+                                        'starts_at': '2022-08-07',
+                                        'ends_at': '2022-08-07',
+                                        'image_url': 'https://asite.com',
+                                        'address': 'c',
+                                        'club': test_club.id
+                                        })
+
+    # then
+    created_event = Events.objects.get(id=response.data['id'])
+
+    assert response.status_code == status.HTTP_201_CREATED
+    assert response.data == {'id': created_event.id,
+                             'name': 'Introduction to Git and Github',
+                             'description': 'some desc',
+                             'starts_at': '2022-08-07',
+                             'ends_at': '2022-08-07',
+                             'registration_required': False,
+                             'created_by': None,
+                             'image_url': 'https://asite.com',
+                             'club': 1,
+                             'registration_approval_required': False,
+                             'address': 'c'}
+    assert created_event.name == 'Introduction to Git and Github'
+
+
+@pytest.mark.django_db
+def testCreateEvents_adminRequest_creationSuccessful(client, test_user, test_club):
+    # given
+    test_user.is_staff = True
```

Comment:
After update, don't we need to save the test_user?

Reply:
> After update, don't we need to save the test_user?

It does needs to be saved according to the docs.